### ISSUE TITLE
Recurring event is being deleted from Outlook

### ISSUE THREAD

**Issue by dlepp** [phw198/OutlookGoogleCalendarSync#1243]
> Checklist:-
> * [x] [Upgrade](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Upgrading) to the latest release (alpha if available) ![latest release](https://img.shields.io/github/release/phw198/OutlookGoogleCalendarSync/all.svg)
> * [x] Reproduce the issue
> * [x] Complete the information below
> * [x] Provide your `OGcalsync.log` file (see [how-to](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-problems#locating-your-ogcalsynclog-file))  

**OGCS Version**: ~~v2.8.7~~ v2.8.4
**Edition**: Portable  
**Sync Direction**:  O<->G

**Bug Description**  
I have deleted two recurring events in Outlook and then recreated them.  They were initially synced to Google, however on the subsequent sync, one event is deleted while the other stays intact.  I recovered the deleted event from Outlook Deleted Items, by opening it and copying to My Calendar, but it is then deleted after each sync.  I have also tried deleting it from google, doing a full one-way sync O->G and then resetting back to O<->G, but it continues to be deleted from Outlook.  The other event for some reason is not affected.  I have attached the log entries for the initial sync when the events are copied to Google, and the subsequent sync when one is deleted. 

**Steps to Reproduce the Issue**  
1. Created recurring event in Outlook
2. Started sync, allowed item to be deleted from Outlook
3. Restore event from Outlook deleted folder
4. Repeat sync

**Any other information**  
Any additional information that may help troubleshoot the issue.  
:paperclip: Attach your `OGcalsync.log` file.

[OGCS_log.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/6786588/OGCS_log.txt)

**Comment by phw198** [comment]
Your logfile shows you running v2.8.4, not v2.8.7. I think your issue was likely resolved in v2.8.6 and #1123.

:arrow_double_up: Please upgrade to ![latest release](https://img.shields.io/github/release/phw198/OutlookGoogleCalendarSync/all.svg) by going to the `About` tab in OGCS, ticking the `Check for Alpha releases` box and then clicking `Check for Update`. 

If this fails for some reason, [download the latest version directly from the project website](https://phw198.github.io/OutlookGoogleCalendarSync/).

You should then be able to force a one way sync O->G to create the item repeatedly deleting item, and then switch back to two-way sync.

**Comment by shiftclickdesign** [comment]
I'm encountering a similar issue in a one-way sync from Outlook to Google using OGCS v2.8.7.0.

Only the current week's instance of a recurring meeting disappears in the Google calendar. If I delete the entire series in GCal and re-sync, it appears. On the very next sync, it disappears.

This recurring meeting was scheduled by someone else, so I don't have control over it. It's likely that the current week's instance has been edited by the sender to tailor meeting notes for the current week. Subsequent instances seem unaffected.

[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/6862323/OGcalsync.log)

**Comment by phw198** [comment]
@shiftclickdesign Your log file is filled with 
```
System.Runtime.InteropServices.COMException: You changed one of the recurrences of this item, and this instance no longer exists. Close any open items and try again.
Code: 0xCE604005,-832552955; Location: get_AppointmentItem() in <Unknown File>
```
Does the problem persist if you:
1. Close down Outlook and OGCS
2. Delete the Gcal series
3. Restart Outlook and OGCS
4. Sync twice

Are you running against an Exchange account? In [cached mode](https://support.microsoft.com/en-us/office/turn-on-cached-exchange-mode-7885af08-9a60-4ec3-850a-e221c1ed0c1c)?

**Comment by shiftclickdesign** [comment]
Yes. The behavior persists even when I’ve followed the steps that you outline below.  

I just discovered that it also happens when *I* (rather than solely the meeting organizer) modify a single occurrence of a recurring event. So the issue is there seemingly when *anyone* modifies a single instance of a recurring event.  

Outlook is running in cached Exchange mode.


From: Paul Woolcock ***@***.******@***.***)
Reply: phw198/OutlookGoogleCalendarSync ***@***.******@***.***)
Date: July 24, 2021 at 3:59:54 PM
To: phw198/OutlookGoogleCalendarSync ***@***.******@***.***)
CC: shiftclickdesign ***@***.******@***.***), Mention ***@***.******@***.***)
Subject: Re: [phw198/OutlookGoogleCalendarSync] Recurring event is being deleted from Outlook (#1243)

**Comment by phw198** [comment]
@shiftclickdesign Please could you upload your ogcalsync.log file for the period covering those test steps, so I can try and trace what's happening

**Comment by shiftclickdesign** [comment]
Log file is attached.

[OGcalsync.2021-07-25.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/6878230/OGcalsync.2021-07-25.log)

**Comment by dlepp** [comment]
Thanks for the feedback on the original issue.  I had upgraded to 2.8.7 before posting and still had the issue, but accidentally uploaded an older logfile.  I have uploaded the latest log - I have set it to ask for deletions.  It is trying to delete 7/9/2021 3:30 PM (R) "TBS meeting" from Outlook despite the event being present in both Outlook and Google.  If I manually 
delete the occurence from Google, it still tries to delete the Outlook event - I would expect it to restore it in Google instead.

[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/6885629/OGcalsync.log)

**Comment by phw198** [comment]
@dlepp The terminology in your opening post is a little confusing - could you reword, taking care to distinguish between a recurring **series**, and an **instance/occurrence** of that recurring series.

If you are deleting either the entire series or just a single occurrence in Google, because you have two-way sync on, of course it will delete the same in Outlook - if it recreated stuff instead, that would certainly be unexpected sync behaviour. I'm pretty sure you just need to [follow these steps](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Hints-&-Tips#fixing-a-stubborn-unwanted-deletion-with-full-sync) to stop an unwanted deletion.

Basically, decide which is the "master" calendar, in the other calendar manually delete the series that is a problem, then force a full one-way sync to recreate it properly, reinstate two-way sync.

**Comment by phw198** [comment]
@shiftclickdesign Your log is still filled with the same error message when trying to open meeting series. For example instances for 10AM "Dialog categories sub group" after 19-Jul seem to be showing the problem - can you try opening these manually in Outlook (selecting "just this one", not the "the series") and see if you get the same error reported directly in the Outlook applicaton?

If you do, Google results seem to say the only option is to have the meeting organiser delete the series and send out a new one! Given, the message says "this instance no longer exists", OGCS assumes it _is_ deleted, hence why syncing a "deletion", but ultimately this seems like some kind of corruption in the series (see here for [someone affected after upgrading to 2016](https://social.technet.microsoft.com/Forums/en-US/fd60ed27-8755-4eae-ae8c-6c7f9245ffdd/issue-with-recurring-meeting-after-upgrade-to-outlook-2016?forum=Office2016setupdeploy)).

**Comment by dlepp** [comment]
It was only an occurence of a recurring series that OGCS was trying to delete from my Outlook calendar.  I had tried deleting the occurence in Google and then doing a one-way sync O>G, but that did not restore the occurence in Google for some reason.  However, deleting the entire series in G and then 1-way O>G has solved it.  Thanks for your help.

**Comment by phw198** [comment]
@shiftclickdesign I'm going to close this ticket now as the OP's issue is resolved. As previously posted, I also suspect your issue is down to your OST file/Outlook, rather than a bug in OGCS.
